File: app/src/main/java/ch/ralena/natibo/adapter/LanguageDetailAdapter.java
package ch.ralena.natibo.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import ch.ralena.natibo.R;
import ch.ralena.natibo.object.Pack;
import io.reactivex.subjects.PublishSubject;
import io.realm.RealmList;

public class LanguageDetailAdapter extends RecyclerView.Adapter<LanguageDetailAdapter.ViewHolder> {

	PublishSubject<Pack> packSubject = PublishSubject.create();

	public PublishSubject<Pack> asObservable() {
		return packSubject;
	}

	private RealmList<Pack> packs;

	public LanguageDetailAdapter(RealmList<Pack> packs) {
		this.packs = packs;
	}

	@NonNull
	@Override
	public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		View view;
		view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_language_detail, parent, false);
		return new ViewHolder(view);
	}

	@Override
	public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
		if (position < getItemCount())
			holder.bindView(packs.get(position));
	}

	@Override
	public int getItemCount() {
		return packs.size();
	}

	class ViewHolder extends RecyclerView.ViewHolder {
		private View view;
		private TextView book;
		private TextView numSentences;
		private Pack pack;

		ViewHolder(View view) {
			super(view);
			this.view = view;
			book = view.findViewById(R.id.packTitleLabel);
			numSentences = view.findViewById(R.id.numSentencesLabel);
			this.view.setOnClickListener(v -> packSubject.onNext(pack));
		}

		void bindView(Pack pack) {
			this.pack = pack;
			book.setText(pack.getBook());
			numSentences.setText("" + pack.getSentences().size());
		}
	}
}

File: app/src/main/java/ch/ralena/natibo/ui/language/detail/adapter/LanguageDetailAdapter.kt
package ch.ralena.natibo.ui.language.detail.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ch.ralena.natibo.R
import ch.ralena.natibo.data.room.`object`.Pack
import ch.ralena.natibo.data.room.`object`.PackRoom
import ch.ralena.natibo.ui.base.BaseRecyclerAdapter
import io.reactivex.subjects.PublishSubject
import io.realm.RealmList
import javax.inject.Inject

class LanguageDetailAdapter @Inject constructor(
	private val packs: ArrayList<PackRoom>
	) :
		BaseRecyclerAdapter<LanguageDetailAdapter.ViewHolder, LanguageDetailAdapter.Listener>() {
	interface Listener {
		fun onLanguagePackClicked(pack: PackRoom)
	}

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		val view = LayoutInflater.from(parent.context).inflate(R.layout.item_language_detail, parent, false)
		return ViewHolder(view)
	}

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		if (position < itemCount) holder.bindView(packs[position])
	}

	override fun getItemCount(): Int {
		return packs.size
	}

	fun loadLanguagePacks(packs: List<PackRoom>) {
		this.packs.clear()
		this.packs.addAll(packs)
		notifyDataSetChanged()
	}

	inner class ViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
		private val book: TextView
		private val numSentences: TextView
		private lateinit var pack: PackRoom

		fun bindView(pack: PackRoom) {
			this.pack = pack
			book.text = pack.name
			// todo show sentence count
//			numSentences.text = pack.sentences.size.toString()
		}

		init {
			book = view.findViewById(R.id.packTitleLabel)
			numSentences = view.findViewById(R.id.numSentencesLabel)
			view.setOnClickListener {
				for (l in listeners)
					l.onLanguagePackClicked(pack)
			}
		}
	}
}